### DISCUSSION TITLE
pycord comparison against Hikari (or other alternatives)

### DISCUSSION THREAD

**Discussion by Znote** [Pycord-Development/pycord#721] [Q&A] [answered]
Shoot, yesterday I began working on my first Discord bot, and I ended up doing the classic `pip install discord`. 

Then I started digging into tutorials, and it came to life that ey, that repo is archieved and set for obsoletion by april. 

Then I found what I believe are 2 good fork candidates, pycord and Hikari. (Any other .py alternatives that bring something to the table)?

**Question is:** 
What are the differences, and what different direction do they wish to go in the future, when they keep developing themselves further away from each other?
(Not asking which is defacto-best, as this is posted on pycord, and there might be a touch of bias due to that). 

I'm also slightly worried about the discord change that requires slash commands, am I wrong to assume the bot won't be able to read channel messages unless explicitly contacted through a command? 

One of the features I wanted to develop is a chat channel that synchronizes between an in-game client chat channel, with a discord server chat channel. (etc make a Help channel reachable and inter-communicative from inside,- and outside of a game im working on). Does this violate terms of service or will cease to work soon?

**Comment by VincentRPS** [comment, marked as answer]
hikari isn’t any where near a fork, it is a whole different project separate from discord.py started before it was archived. So following discord.py tutorials will not teach you hikari.

**Reply by VincentRPS** [reply]
hikari does focus more on modularity and customizability more than what discord.py does, they have hundreds if not thousands of abcs plus more, while discord.py and the forks of it are more user friendly, e.g ext.commands is a command handler, a thing most discord library’s python or not don’t have stock.

**Reply by VincentRPS** [reply]
Also another thing you might care about is hikari doesn’t come with support for voice stuff while discord.py a related forks do.

**Reply by VincentRPS** [reply]
In the sphere of app commands support Pycord and hikari both support them.

**Comment by Lulalaby** [comment]
> One of the features I wanted to develop is a chat channel that synchronizes between an in-game client chat channel, with a discord server chat channel. (etc make a Help channel reachable and inter-communicative from inside,- and outside of a game im working on). Does this violate terms of service or will cease to work soon?

It does not violate. If your bot is verified, request the intent for this usecase. If not, just enable it in the developer portal.
